cpp/open3d/core/Tensor.h

```diff
@@ -219,6 +219,104 @@ class Tensor {
                        Dtype dtype,
                        const Device& device = Device("CPU:0"));
 
+    /// Create a 1-D tensor with given list.
+    /// For example,
+    /// core::Tensor::Init<float>({1,2,3});
+    template <typename T>
+    static Tensor Init(const std::initializer_list<T> in_list,
+                       const Device& device = Device("CPU:0")) {
+        Dtype type = Dtype::FromType<T>();
+        std::vector<T> ele_list;
+        ele_list.insert(ele_list.end(), in_list.begin(), in_list.end());
+
+        SizeVector shape{static_cast<int64_t>(in_list.size())};
+        return Tensor(ele_list, shape, type, device);
+    };
+
+    /// Create a 2-D tensor with given list.
+    /// For example,
+    /// core::Tensor::Init<float>({{1,2,3},{4,5,6}});
+    template <typename T>
+    static Tensor Init(
+            const std::initializer_list<std::initializer_list<T>> in_list,
+            const Device& device = Device("CPU:0")) {
+        Dtype type = Dtype::FromType<T>();
+        std::vector<T> ele_list;
+        int64_t dim0_size = static_cast<int64_t>(in_list.size());
+        int64_t dim1_size = -1;
+        for (const auto& ele0 : in_list) {
+            if (dim1_size == -1) {
+                dim1_size = static_cast<int64_t>(ele0.size());
+            } else {
+                if (static_cast<int64_t>(ele0.size()) != dim1_size) {
+                    utility::LogError(
+                            "Cannot create Tensor with ragged nested "
+                            "sequences( nested"
+                            "lists with unequal sizes or shapes).");
+                }
+            }
+            ele_list.insert(ele_list.end(), ele0.begin(), ele0.end());
+        }
+
+        SizeVector shape{dim0_size, dim1_size};
+        return Tensor(ele_list, shape, type, device);
```

Comment:
Can you check for zero size here too (similar to line 308)? eg `Init({{}})`: size = (1,0) convert to (1,)

Reply:
Yes. Let's add a unit test for that.

In numpy, this is:
```python
In [5]: np.array([[]]).shape
Out[5]: (1, 0)
```

Reply:
I checked for this case and all the 0-sized initialiser list inputs are being passed to the function for 3-D. I added a test with "core::Tensor::Init<bool>({{}, {}})" for the 2-D case. I also added test with "core::Tensor::Init<bool>({{}})"